projects/aca-shared/src/lib/adf-extensions/extensions-data-loader.guard.spec.ts

```diff
@@ -131,5 +131,32 @@ describe('ExtensionsDataLoaderGuard', () => {
       expect(erroredSpy).not.toHaveBeenCalled();
       expect(completedSpy).toHaveBeenCalled();
     });
+
+    it('should call canActivate only once', () => {
+      const subject1 = new Subject<true>();
+      const extensionLoaders = {
+        fct1: function() {
+          return subject1.asObservable();
+        }
+      };
+      const extensionLoaderSpy = spyOn(extensionLoaders, 'fct1');
+      const guard = new ExtensionsDataLoaderGuard([extensionLoaders.fct1]);
+
+      subject1.next();
+      expect(guard.involved).toEqual(false);
```

Comment:
This needs to be private. Unit test needs to test the behaviour of the function. This way, you are doing nothing else, but testing the implementation, which is something which we should always avoid.